### PR TITLE
[Security] Run check jwt before all statecode api routes

### PR DESCRIPTION
**Pull request by Catacola**
## Description of the change

Updates the server to call the JWT middleware before trying to check the state code.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes [#XXXX]

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: server/utils/__tests__/validateStateCode.test.js [hunk 1]
```diff
@@ -55,7 +54,7 @@ const expectStatus = (req, status) => {
     }),
   };
 
-  const validator = validateStateCode(exemptRoutes);
+  const validator = validateStateCode();
```
Comment (jovergaag) [inline server/utils/__tests__/validateStateCode.test.js:57]: nit: This is redundant.

File: server/utils/__tests__/validateStateCode.test.js [hunk 2]
```diff
@@ -114,28 +113,4 @@ describe("validateStateCode", () => {
...
-
-  it("should return 400 if /api/ is not followed by a state code", () => {
-    const req = makeReq("/api/some/other/path", "some", "not_some");
-    expectStatus(req, 400);
-  });
-
-  it("should return 400 if /api/ is not followed by a properly formatted state code", () => {
-    const req = makeReq("/api/us_xxx/other/path", "some", "not_some");
-    expectStatus(req, 400);
-  });
```
Comment (jovergaag) [inline server/utils/__tests__/validateStateCode.test.js:140]: Even though this functionality is removed from `validateStateCode`, should we move the tests somewhere else? Maybe `server.test.js`? Makes me nervous not testing these scenarios.

File: server/app.js [hunk 3]
```diff
@@ -53,6 +54,15 @@ const isDemoMode = process.env.IS_DEMO === "true";
 
 const authEnv = process.env.AUTH_ENV;
 
+const stateApiBaseRoute = "/api/:stateCode(us_[a-z][a-z]|US_[A-Z][A-Z])/";
+
+const routesExemptFromJwtValidation = [
```
Comment (jovergaag) [inline server/app.js:59]: These are exempt from both JWT and state code validation, maybe change the name to reflect that?

**Review by jovergaag — CHANGES_REQUESTED** [review]

File: server/utils/__tests__/validateStateCode.test.js (hunk 2 above)
Comment (Catacola) [reply server/utils/__tests__/validateStateCode.test.js:140]: I've updated the PR to test the us_xxx case. There was an existing test to check for using "HI" as a state code which covers the 3rd case.

The exempt ones (cases 1 and 2) were already being checked by the tests on refreshCache. In those tests, demo mode is off, so it's trying to check the jwt/state code, but refreshCache is an exempted route, so the request succeeds.

File: server/app.js (hunk 3 above)
Comment (Catacola) [reply server/app.js:59]: I've created a second variable that explicitly lists routes exempt from state code validation which have a state code in their path.

**Review by jovergaag — APPROVED** [review]

**Pull request merged**